core/src/sumeragi/fault.rs

```diff
@@ -310,1495 +400,970 @@ pub fn run_sumeragi_main_loop<F>(
-
-                                    if let Err(error) = wsv_guard.apply(block.clone()) {
-                                        panic!("Failed to apply block on WSV. This is absolutely not acceptable.");
-                                    }
-
-                                    for event in Vec::<Event>::from(&block) {
-                                        trace!(?event);
-                                        drop(sumeragi.events_sender.send(event));
-                                    }
-
-                                    state_machine_guard.latest_block_height = block.header().height;
-                                    state_machine_guard.latest_block_hash = block.hash();
-
-                                    let previous_role = state_machine_guard
-                                        .current_topology
-                                        .role(&sumeragi.peer_id);
-                                    state_machine_guard.current_topology.apply_block(block_hash);
-                                    info!(
-                                        prev_peer_role = ?previous_role,
-                                        new_peer_role = ?state_machine_guard.current_topology.role(&sumeragi.peer_id),
-                                        new_block_height = %state_machine_guard.latest_block_height,
-                                        %block_hash,
-                                        "Committing block"
-                                    );
-                                    sumeragi.kura.blocking_store_block(block);
-                                    SumeragiWithFault::<F>::update_network_topology(
-                                        &mut state_machine_guard.current_topology,
-                                        &wsv_guard,
-                                    );
-                                    has_sent_transactions = false;
-                                    voting_block_option = None;
-                                    println!(
-                                        "Observing peer has committed a valid block it was given."
-                                    );
-                                }
-                            }
+            if has_sent_transactions && sent_transaction_time.elapsed() > sumeragi.pipeline_time() {
+                trace!("Suspecting all peers for not producing a block with my transaction.");
+                request_view_change(
+                    sumeragi,
+                    &mut state_machine_guard,
+                    &mut view_change_proof_chain,
+                    current_view_change_index,
+                );
+                sent_transaction_time = Instant::now();
+            }
+        }
+
+        if state_machine_guard.current_topology.role(&sumeragi.peer_id) == Role::ObservingPeer {
+            if maybe_incoming_message.is_some() {
+                let incoming_message = maybe_incoming_message.take().unwrap();
```

Comment:
yes

Reply:
Well no actually, non of the changes to `if let` you suggest actually work because they get killed by the borrow checker. It is written `is_some` followed by an unwrap because that doesn't consume the value.

Reply:
I think `Option::take` would help in this case if value is `None` this is no-op anyway.

Reply:
That might be so. How it is finally written isn't really that important to me. You could go and change all these things to the way you prefer after we've merged. What matters to me is getting things done faster. `if a.is_some() { a.unwrap() ...` and `if let` do the exact same thing. So changing from one to the other is shuffling papers around on the desk without actually changing anything for the end user.

Reply:
So i agree with you that this 2 approaches do the same thing.
However i think that such stylistic choices go against our guidelines and because of this we patch then with `allow` attributes.

>  without actually changing anything for the end user

Beside end users there developers who actually read and write code. 
I believe that such concessions tend to accumulate and it easier to fix them sooner than later.

I think we need third opinion on this one: @appetrosyan what do you think?

Reply:
@SamHSmith 's style needs to be reworked anyway. I'm thinking that this is better done after we have a safe-to-work-with Sumeragi in staging.